trurl's checksrc.pl is outdated

`trurl`'s and `curl`'s `checksrc.pl` are not synchronised, so `trurl` is still using an older version.

Running `curl`'s `checksrc.pl` on `trurl`'s source files, finds those two problems:
```
$ ../curl/scripts/checksrc.pl trurl.c version.h
./trurl.c:689:9: warning: if with space (SPACEBEFOREPAREN)
       if (*i < 32)
         ^
./trurl.c:747:11: warning: not indented 2 steps (uses 4) (INDENTATION)
           continue;
           ^
checksrc: 0 errors and 2 warnings
checksrc: 0 errors and 1 warnings suppressed
```

--- a/checksrc.pl
+++ b/checksrc.pl
@@ -90,6 +90,7 @@ my %warnings = (
     'SPACEBEFORECOMMA' => 'space before a comma',
     'SPACEBEFOREPAREN' => 'space before an open parenthesis',
     'SPACESEMICOLON'   => 'space before semicolon',
+    'SPACESWITCHCOLON' => 'space before colon of switch label',
     'TABS'             => 'TAB characters not allowed',
     'TRAILINGSPACE'    => 'Trailing whitespace on the line',
     'TYPEDEFSTRUCT'    => 'typedefed struct',
@@ -517,7 +518,8 @@ sub scanfile {
 
         my $nostr = nostrings($l);
         # check spaces after for/if/while/function call
-        if($nostr =~ /^(.*)(for|if|while| ([a-zA-Z0-9_]+)) \((.)/) {
+        if($nostr =~ /^(.*)(for|if|while|switch| ([a-zA-Z0-9_]+)) \((.)/) {
+            my ($leading, $word, $extra, $first)=($1,$2,$3,$4);
             if($1 =~ / *\#/) {
                 # this is a #if, treat it differently
             }
@@ -527,15 +529,16 @@ sub scanfile {
             elsif(defined $3 && $3 eq "case") {
                 # case must have a space
             }
-            elsif($4 eq "*") {
-                # (* beginning makes the space OK!
+            elsif(($first eq "*") && ($word !~ /(for|if|while|switch)/)) {
+                # A "(*" beginning makes the space OK because it wants to
+                # allow funcion pointer declared
             }
             elsif($1 =~ / *typedef/) {
                 # typedefs can use space-paren
             }
             else {
-                checkwarn("SPACEBEFOREPAREN", $line, length($1)+length($2), $file, $l,
-                          "$2 with space");
+                checkwarn("SPACEBEFOREPAREN", $line, length($leading)+length($word), $file, $l,
+                          "$word with space");
             }
         }
         # check for '== NULL' in if/while conditions but not if the thing on
@@ -688,6 +691,12 @@ sub scanfile {
                       $line, length($1), $file, $ol, "no space before semicolon");
         }
 
+        # check for space before the colon in a switch label
+        if($l =~ /^( *(case .+|default)) :/) {
+            checkwarn("SPACESWITCHCOLON",
+                      $line, length($1), $file, $ol, "no space before colon of switch label");
+        }
+
         # scan for use of banned functions
         if($l =~ /^(.*\W)
                    (gmtime|localtime|
@@ -752,6 +761,32 @@ sub scanfile {
                     checkwarn("INDENTATION", $line, length($1), $file, $ol,
                               "not indented $indent steps (uses $diff)");
 
+                }
+            }
+        }
+
+        # if the previous line starts with if/while/for AND ends with a closed
+        # parenthesis and there's an equal number of open and closed
+        # parentheses, check that this line is indented $indent more steps, if
+        # not a cpp line
+        elsif(!$prevp && ($prevl =~ /^( *)(if|while|for)(\(.*\))\z/)) {
+            my $first = length($1);
+            my $op = $3;
+            my $cl = $3;
+
+            $op =~ s/[^(]//g;
+            $cl =~ s/[^)]//g;
+
+            if(length($op) == length($cl)) {
+                # this line has some character besides spaces
+                if($l =~ /^( *)[^ ]/) {
+                    my $second = length($1);
+                    my $expect = $first+$indent;
+                    if($expect != $second) {
+                        my $diff = $second - $first;
+                        checkwarn("INDENTATION", $line, length($1), $file, $ol,
+                                  "not indented $indent steps (uses $diff)");
+                    }
                 }
             }
         }

--- a/trurl.c
+++ b/trurl.c
@@ -685,7 +685,7 @@ static void jsonString(FILE *stream, const char *in, size_t len,
       fputs("\\t", stream);
       break;
     default:
-      if (*i < 32)
+      if(*i < 32)
         fprintf(stream, "u%04x", *i);
       else {
         char out = *i;
@@ -743,7 +743,7 @@ static void json(struct option *o, CURLU *uh)
 
       /* don't print out empty/trimmed values */
       if(!qpairsdec[j][0])
-          continue;
+        continue;
       if(!first)
         fputs(",\n", stdout);
       first = false;

